ImperatorToCK3/Source/CK3/CK3World.cpp

```diff
@@ -245,22 +250,110 @@ void CK3::World::linkCountiesToTitleHolders(const ImperatorWorld::World& sourceW
+						if (impMonarch >= 0) countyTitle->holder = "imperator" + std::to_string(impMonarch);
+					}
+					else // county is probably outside of Imperator map
+					{
+						auto vanillaHistory = titlesHistory.popTitleHistory(name);
+						if (vanillaHistory) countyTitle->historyString = *vanillaHistory;
+					}
 				}
-				else // county is probably outside of Imperator map
+			}
+			titles.insert(std::pair(name, countyTitle));
+		}
+	}
+}
+
+
+void CK3::World::insertVanillaNonCountiesToTitles(const ImperatorWorld::World& impWorld)
+{
+	for (const auto& [name, landedTitle] : landedTitles.getFoundTitles())
+	{
+		if (name.find("c_") != 0 && name.find("b_") != 0 ) // title is a duchy or higher
+		{
+			bool toInsert = true;
+			// important check: if any of the title's de jure counties's holder is "0", don't insert the title
+			for (const auto& [vassalTitleName, deJureVassal] : landedTitle.getFoundTitles())
+			{
+				if (vassalTitleName.find("c_")==0 && titles.count(vassalTitleName) && titles[vassalTitleName]->holder == "0")
 				{
-					auto vanillaHistory = titlesHistory.popTitleHistory(name);
-					if (vanillaHistory) countyTitle->historyString = *vanillaHistory;
+					toInsert = false;
+					break;
 				}
 			}
-			titles.insert(std::pair(name, countyTitle));
+
+			if(toInsert)
+			{
+				auto vanillaTitle = std::make_shared<Title>();
+				vanillaTitle->titleName = name;
+				auto vanillaHistory = titlesHistory.popTitleHistory(name);
+				if (vanillaHistory) vanillaTitle->historyString = *vanillaHistory;
+				titles.insert(std::pair(name, vanillaTitle));
+			}
 		}
 	}
 }
+
+void CK3::World::linkSpouses(const ImperatorWorld::World& impWorld)
+{
+	auto counterSpouse = 0;
+	for (const auto& [ck3CharacterID, ck3Character] : characters)
+	{
+		std::map<int, std::shared_ptr<Character>> newSpouses;
+		// make links between Imperator characters
+		for (auto [impSpouseID, impSpouseCharacter] : ck3Character->imperatorCharacter->getSpouses())
+		{
+			if(impSpouseCharacter!=nullptr)
+			{
+				auto ck3SpouseCharacter = impSpouseCharacter->getCK3Character();
+				ck3Character->addSpouse(std::pair(ck3SpouseCharacter->ID, ck3SpouseCharacter));
+				ck3SpouseCharacter->addSpouse(std::pair(ck3CharacterID, ck3Character));
+				++counterSpouse;
+			}
+		}
+	}
+	Log(LogLevel::Info) << "<> " << counterSpouse << " spouses linked in CK3.";
+}
+
+
+void CK3::World::linkMothersAndFathers(const ImperatorWorld::World& impWorld)
+{
+	auto counterMother = 0;
+	auto counterFather = 0;
+	for (const auto& [ck3CharacterID, ck3Character] : characters)
+	{
+		// make links between Imperator characters
+		auto [impMotherID, impMotherCharacter] = ck3Character->imperatorCharacter->getMother();
+		if (impMotherCharacter != nullptr)
+		{
+			auto ck3MotherCharacter = impMotherCharacter->getCK3Character();
+			ck3Character->setMother(std::pair(ck3MotherCharacter->ID, ck3MotherCharacter));
```

Comment:
not impMotherID?

Reply:
No, the CK3 ID, which is different.

Reply:
I see what's happening now. All is well.